benchexec/container.py

```diff
@@ -335,13 +337,30 @@ def setup_user_mapping(
         # but if it does not exist, we do not need to write to it.
         if e.errno != errno.ENOENT:
             logging.warning("Could not write to setgroups file in /proc: %s", e)
+            exception_occurred = True
 
     try:
         # map gid internally to our gid externally
         gid_map = f"{gid} {parent_gid} 1"
         util.write_file(gid_map, proc_child, "gid_map")
     except OSError as e:
         logging.warning("Creating GID mapping into container failed: %s", e)
+        exception_occurred = True
+
+    # Ubuntu 24.04 restricts user namespaces, output error message here
+    if (
+        exception_occurred
+        and util.check_ubuntu_version() == "24.04"
```

Comment:
I think we can skip this check. Future Ubuntu versions will likely behave the same, and other distributions might also adopt the behavior from Ubuntu. So just check whether `apparmor_restrict_unprivileged_userns` is enabled, and the error message can then tell people do disable it, no matter what the distribution is.

Reply:
Agree with that. btw, is there a way to locally deploy these CI checks? I want to be able to run them before pushing to avoid excessive commits :)

Reply:
Yes, you can basically run everything locally if you install the tools like Black etc., have a look in `.gitlab-ci.yml` for what exactly gets executed.

Note that running the tests in all combinations is more tricky. The CI tests with cgroups v1, if you test locally you will probably have cgroups v2. But we want to have it work with all versions anyway. :-)

Reply:
Please review if there are any other modifications needed. I have already skipped the Ubuntu version check.